hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelation.java

```diff
@@ -0,0 +1,95 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.hxi_extension.rest.api;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.alfresco.hxi_connector.hxi_extension.rest.api.exception.AgentAvatarException;
+import org.alfresco.rest.framework.WebApiDescription;
+import org.alfresco.rest.framework.core.exceptions.NotFoundException;
+import org.alfresco.rest.framework.core.exceptions.RelationshipResourceNotFoundException;
+import org.alfresco.rest.framework.resource.RelationshipResource;
+import org.alfresco.rest.framework.resource.actions.interfaces.RelationshipResourceAction;
+import org.alfresco.rest.framework.resource.content.BinaryResource;
+import org.alfresco.rest.framework.resource.content.FileBinaryResource;
+import org.alfresco.rest.framework.resource.parameters.Parameters;
+import org.alfresco.util.TempFileProvider;
+
+@Slf4j
+@AllArgsConstructor
+@RelationshipResource(name = "avatars", title = "Avatars of agents", entityResource = AgentsEntityResource.class)
+public class AvatarRelation implements RelationshipResourceAction.ReadById<BinaryResource>
+{
+    private HttpClient client;
+
+    @Override
+    @WebApiDescription(title = "Get Agent Avatar image")
+    public BinaryResource readById(String agentId, String avatarId, Parameters parameters) throws RelationshipResourceNotFoundException
+    {
+        if (!avatarId.equals("-default-"))
+        {
+            log.info("Avatar id is different than -default-");
+            throw new NotFoundException(String.format("Avatar with id=%s not found", avatarId));
+        }
+
+        return getSampleAvatar();
+    }
+
+    private FileBinaryResource getSampleAvatar()
+    {
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create("https://www.wikipedia.org/portal/wikipedia.org/assets/img/Wikipedia-logo-v2@2x.png"))
+                .GET()
+                .build();
+
+        try
+        {
+            HttpResponse<InputStream> httpResponse = client.send(request, HttpResponse.BodyHandlers.ofInputStream());
+            File file = TempFileProvider.createTempFile(httpResponse.body(), "RenditionsApi-", ".png");
+            return new FileBinaryResource(file, null);
+        }
+        catch (IOException | InterruptedException e)
+        {
+            log.error("Error getting avatar image.", e);
+            throw new AgentAvatarException("Avatar image cannot be received.", e);
```

Comment:
The last time we discussed this in the team we agreed upon 503 for this issue:
```
throw new WebScriptException(SC_SERVICE_UNAVAILABLE, "Failed to ask question", e);
```

I noticed that we have no consistency over it at the moment though, so I'll create a ticket to fix it in all endpoints.

Reply:
More generally it would be better to put this code in `HxInsightClient`. Have a look at what we're doing for the other endpoints.

Reply:
Updated